Vungle/src/main/java/com/mopub/mobileads/VungleRouter.java

```diff
@@ -70,32 +72,37 @@ public void onResume(@NonNull final Activity activity) {
         private final String placementId;
         @Nullable
         private final String adMarkup;
+        @NonNull
+        private AdSize adSize = AdSize.VUNGLE_DEFAULT;
 
         public AdRequest(@NonNull String placementId, @Nullable String adMarkup) {
             this.placementId = placementId;
             this.adMarkup = adMarkup;
         }
 
-        @Override
-        public int hashCode() {
-            int result = placementId.hashCode();
-            result = 31 * result + (adMarkup != null ? adMarkup.hashCode() : 0);
-            return result;
+        public AdRequest(@NonNull String placementId, @Nullable String adMarkup, @NonNull AdSize adSize) {
```

Comment:
I know it's kind of copied from Vungle SDK, but why do we considering AdRequest for the same admarkup to be different for different adSizes?

Reply:
I would propose to get rid of sWaitingList, and give all control to Vungle SDK to handle differences in adSizes and Markups.
Just by putting everything to Runnable which is called once Vunglr SDK is inited

Reply:
Ideally mediation should never send multiple loadAd with adMarkup for same eventId, so this should not really matter. Moreover AdRequest object is used only during initialization
@alexander-gubov Is there any specific usecase you can think of where ignoring the AdSize would be beneficial ?

Reply:
Nope, it's about simplification of the logic:
instead of having AdRequest we can just have "Router".init(InitCallback callback) call which will handle everything for the cases wen Vungle is not inited (AdMob adapter has such logic already).
This way we don't need AdRequest knowing about adsize at all and could get rid of AdRequest completely: init callback will store final "AdMarkup" and "AdSize" passed to Banner/Vungle#loadAd when or if Vungle is already inited